arbor/memory/device_coordinator.hpp

```diff
@@ -115,9 +122,10 @@ class device_coordinator {
     using rebind = device_coordinator<Tother, Allocator>;
 
     view_type allocate(size_type n) {
-        Allocator allocator;
-
-        pointer ptr = n>0 ? allocator.allocate(n) : nullptr;
+        pointer ptr = nullptr;
+        if (n > 0) {
+            ptr = Allocator{}.allocate(n);
+        }
```

Comment:
This looks semantically almost identical. I honestly don't understand the original complaint from the static analyzer:
```
arbor/memory/device_coordinator.hpp:120 V623 Consider inspecting the '?:' operator. A temporary object of the 'array_view' type is being created and subsequently destroyed. Check third operand.                                                            
```
The third argument is `nullptr`.

I think it may be being confused by the final return statement (line 128 of original file).

Reply:
Yeah, let's go with the original here.